code/datums/components/proximity_monitor.dm

```diff
@@ -1,139 +1,472 @@
+	return TRUE
+
+/**
+ * Base proc. Checks if `AM` can pass the edge field checker.
+ *
+ * Arguments:
+ * * atom/movable/AM - the atom trying to pass the edge field checker object
+ * * obj/effect/abstract/proximity_checker/advanced/edge_field/F - the proximity checker object `AM` is trying to pass
+ * * turf/entering - the turf `AM` is entering from
+ */
+/datum/component/proximity_monitor/advanced/proc/edge_field_canpass(atom/movable/AM, obj/effect/abstract/proximity_checker/advanced/edge_field/F, turf/entering)
+	return TRUE
+
+/**
+ * Base proc. Called when something crosses an edge field checker.
+ *
+ * Arguments:
+ * * atom/movable/AM - the atom crossing the edge field checker object
+ * * obj/effect/abstract/proximity_checker/advanced/edge_field/F - the proximity checker object `AM` getting crossed
+ */
+/datum/component/proximity_monitor/advanced/proc/edge_field_crossed(atom/movable/AM, obj/effect/abstract/proximity_checker/advanced/edge_field/F)
+	return TRUE
+
+/**
+ * Base proc. Called when something uncrosses an edge field checker.
+ *
+ * Arguments:
+ * * atom/movable/AM - the atom uncrossing the edge field checker object
+ * * obj/effect/abstract/proximity_checker/advanced/edge_field/F - the proximity checker object `AM` getting uncrossed
+ */
+/datum/component/proximity_monitor/advanced/proc/edge_field_uncrossed(atom/movable/AM, obj/effect/abstract/proximity_checker/advanced/edge_field/F)
+	return TRUE
+
+
+/**
+ * # Basic Proximity Checker
+ *
+ * Inteded for use with the proximity checker component [/datum/component/proximity_monitor].
  * Whenever a movable atom crosses this object, it calls `HasProximity()` on the object which is listening for proximity (`hasprox_receiver`).
  */
 /obj/effect/abstract/proximity_checker
-	name = "Proximity checker"
+	name = "proximity checker"
+	/// The component that this object is in use with, and that will receive `HasProximity()` calls.
+	var/datum/component/proximity_monitor/monitor
 	/// Whether or not the proximity checker is listening for things crossing it.
 	var/active
-	/// The linked atom which has the proximity_monitor component, and will recieve the `HasProximity()` calls.
-	var/atom/hasprox_receiver
 
 // If this object is initialized without a `_hasprox_receiver` arg, it is qdel'd.
-/obj/effect/abstract/proximity_checker/Initialize(mapload, atom/_hasprox_receiver)
-	if(_hasprox_receiver)
-		hasprox_receiver = _hasprox_receiver
-		RegisterSignal(hasprox_receiver, COMSIG_PARENT_QDELETING, .proc/OnParentDeletion)
-		if(isturf(hasprox_receiver.loc)) // if the reciever is inside a locker/crate/etc, they don't detect proximity
-			active = TRUE
-	else
-		stack_trace("/obj/effect/abstract/proximity_checker created without a receiver")
+/obj/effect/abstract/proximity_checker/Initialize(mapload, atom/_hasprox_receiver, datum/component/proximity_monitor/P)
+	if(!_hasprox_receiver)
```

Comment:
Is this still required? Seeing it is not used and only used to see if the creator has it's stuff in order?

Reply:
Yeah it's probably not needed.